pkg/api/v1beta1/defaults.go

```diff
@@ -102,3 +108,14 @@ func init() {
 		},
 	)
 }
+
+// With host networking default all container ports to host ports.
+func defaultHostNetworkPorts(containers *[]Container) {
+	for i := range *containers {
+		for j := range (*containers)[i].Ports {
+			if (*containers)[i].Ports[j].ContainerPort == 0 {
+				(*containers)[i].Ports[j].ContainerPort = (*containers)[i].Ports[j].HostPort
```

Comment:
I prefer the simpler semantic - ContainerPort is always required (in as
much as the whole structure is optional), and iff HostNetwork then default
HostPort = ContainerPort.  Have to handle the case of a user setting
hostport != containerport.

On Mon, Mar 23, 2015 at 11:43 PM, Abhi Shah <notifications@github.com>
wrote:

> In pkg/api/v1beta1/defaults.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5814#discussion_r27005890>
> :
>
> > @@ -102,3 +108,14 @@ func init() {
> >  		},
> >  	)
> >  }
> > +
> > +// With host networking default all container ports to host ports.
> > +func defaultHostNetworkPorts(containers *[]Container) {
> > +	for i := range *containers {
> > +		for j := range (*containers)[i].Ports {
> > +			if (*containers)[i].Ports[j].ContainerPort == 0 {
> > +				(*containers)[i].Ports[j].ContainerPort = (*containers)[i].Ports[j].HostPort
>
> We changed it so that ContainerPort is not required if HostNetwork=true.
>
> What I am trying to say is that, when a user defines ports in the
> HostNetwork=true scenario,
> they still need to define which HostIP they plan to use. So, since they
> are specifying HostIP, i thought it was reasonable to expect that they
> specify HostPort as well.
> ContainerPort can be considered optional in this case because, there is no
> container network namespace.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5814/files#r27005890>
> .
>